src/crimson/osd/main.cc

```diff
@@ -47,7 +47,7 @@ int main(int argc, char* argv[])
 
   args.insert(begin(args), argv[0]);
   try {
-    return app.run_deprecated(args.size(), const_cast<char**>(args.data()), [&] {
+    return app.run(args.size(), const_cast<char**>(args.data()), [&] {
```

Comment:
why?

Reply:
crimson-osd --mkfs need engine exit, run_deprecated can't exit.

Reply:
please note, `run()` calls `run_deprecated()`. the reason i used `run_deprecated()` was to catch and handle the exception thrown by the continuation passed to of `run_deprecated()` by ourselves instead of letting `or_terminate()` to do so.